sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobPartitionManager.java

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.checkpointstore.blob;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.eventhubs.EventProcessor;
+import com.azure.messaging.eventhubs.PartitionManager;
+import com.azure.messaging.eventhubs.models.Checkpoint;
+import com.azure.messaging.eventhubs.models.PartitionOwnership;
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.ContainerAsyncClient;
+import com.azure.storage.blob.models.BlobAccessConditions;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.models.BlobListDetails;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.ListBlobsOptions;
+import com.azure.storage.blob.models.Metadata;
+import com.azure.storage.blob.models.ModifiedAccessConditions;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * Implementation of {@link PartitionManager} that uses
+ * <a href="https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction#blob-storage">Storage Blobs</a>
+ * for persisting partition ownership and checkpoint information. {@link EventProcessor EventProcessors} can use this
+ * implementation to load balance and update checkpoints.
+ *
+ * @see EventProcessor
+ */
+public class BlobPartitionManager implements PartitionManager {
+
+    // All languages should use the same key for metadata. Discuss name for these.
+    private static final String SEQUENCE_NUMBER = "sequenceNumber";
+    private static final String OFFSET = "offset";
+    private static final String OWNER_ID = "ownerId";
+    private static final String ETAG = "eTag";
+
+    private static final String BLOB_PATH_SEPARATOR = "/";
+    private static final ByteBuffer UPLOAD_DATA = ByteBuffer.wrap("".getBytes(UTF_8));
```

Comment:
I thought I remembered Jeff saying that as long as the ETag changes when we updated the metadata, we didn't need to have anything in the blob body on update? Or is this only used on first creation? @chradek

Reply:
Yeah, this is used only on first creation. All subsequent updates will only be for the metadata.